libs/template/resolver.go

```diff
@@ -0,0 +1,97 @@
+package template
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/databricks/cli/libs/git"
+)
+
+type Resolver struct {
+	// One of the following three:
+	// 1. Path to a local template directory.
+	// 2. URL to a Git repository containing a template.
+	// 3. Name of a built-in template.
+	TemplatePathOrUrl string
+
+	// Path to a JSON file containing the configuration values to be used for
+	// template initialization.
+	ConfigFile string
+
+	// Directory to write the initialized template to.
+	OutputDir string
+
+	// Directory path within a Git repository containing the template.
+	TemplateDir string
+
+	// Git tag or branch to download the template from. Only one of these can be
+	// specified.
+	Tag    string
+	Branch string
+}
+
+var ErrCustomSelected = errors.New("custom template selected")
+
+// Configures the reader and the writer for template and returns
+// a handle to the template.
+// Prompts the user if needed.
+func (r Resolver) Resolve(ctx context.Context) (*Template, error) {
+	if r.Tag != "" && r.Branch != "" {
+		return nil, fmt.Errorf("only one of --tag or --branch can be specified")
+	}
+
+	// Git ref to use for template initialization
+	ref := r.Branch
+	if r.Tag != "" {
+		ref = r.Tag
+	}
+
+	var err error
+	var templateName TemplateName
+
+	if r.TemplatePathOrUrl == "" {
+		// Prompt the user to select a template
+		// if a template path or URL is not provided.
+		templateName, err = SelectTemplate(ctx)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		templateName = TemplateName(r.TemplatePathOrUrl)
+	}
+
+	// User should not directly select "custom..." and instead should provide the
+	// file path or the Git URL for the template directly.
```

Comment:
Do you know why it is one of the options to begin with?

Reply:
It was meant as documentation mostly to inform users who run the command on how to setup and use their custom templates:
```
.venv➜  cli git:(refactor-bundle-init-squashed) databricks bundle init       
Using workspace profile: dogfood
Search: █
Please specify a path or Git repository to use a custom template.
See https://docs.databricks.com/en/dev-tools/bundles/templates.html to learn more about custom templates.
```

Reply:
ref PR: https://github.com/databricks/cli/pull/1052